keyboards/melgeek/z70ultra/z70ultra.c

```diff
@@ -0,0 +1,152 @@
+                             },
+                             {4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4}};
+
+const is31_led g_is31_indicator_leds[6] = {
+    {0, 5, 1, 3, 1, 4, 1}, /* RGB71 */
+    {0, 5, 2, 3, 2, 4, 2}, /* RGB72 */
+    {0, 5, 3, 3, 3, 4, 3}, /* RGB73 */
+    {0, 5, 4, 3, 4, 4, 4}, /* RGB74 */
+    {0, 5, 5, 3, 5, 4, 5}, /* RGB75 */
+    {0, 5, 6, 3, 6, 4, 6}, /* RGB76 */
+};
+
+bool led_update_kb(led_t led_state) {
+    if (led_update_user(led_state)) {
+        if (led_state.caps_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[2], 0xff, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[3], 0xff, 0x00, 0x00);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[2], 0x00, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[3], 0x00, 0x00, 0x00);
+        }
+
+        if (led_state.num_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[1], 0x00, 0xff, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[4], 0x00, 0xff, 0x00);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[1], 0x00, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[4], 0x00, 0x00, 0x00);
+        }
+
+        if (led_state.scroll_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[0], 0x00, 0x00, 0xff);
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[5], 0x00, 0x00, 0xff);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[0], 0x00, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[5], 0x00, 0x00, 0x00);
+        }
+
+        // flush the indicator to the buffer
+        IS31FL3741_update_pwm_buffers(DRIVER_ADDR_1, DRIVER_ADDR_1);
+    }
+    return true;
+}
+
+void matrix_init_kb(void) {
+    for (int i = 0; i < DRIVER_INDICATOR_LED_TOTAL; ++i) {
+        is31_led led = g_is31_indicator_leds[i];
+
+        IS31FL3741_set_scaling_registers(&led, 0xFF, 0xFF, 0xFF);
+    }
+
+    IS31FL3741_update_led_control_registers(DRIVER_ADDR_1, 0);
+
+    matrix_init_user();
+}
```

Comment:
Is there a reason that this is being handled here, rather than by `rgb_matrix_indicators_kb`?

Reply:
for z70ultra, the 6 indicator leds (two leds for Capslock, two for Numlock, and two for ScrollLock) not belong to the rgb matrix, I can't properly put the 6 leds into the rgb matrix. you can refer to the g_led_config.

Reply:
Why can't they?  

They don't have to be attached to a switch.  For instance, the massdrop boards have a dozen or so LEDs in the rgb matrix that aren't connected to a switch, and are actually used for underglow, instead.   They are still in the g_led_config, but not in the first part of the array (which defines the associate of keys and LEDs).

Additionally, the corne (crkbd) does the same with the 6 underglow LEDs.  

So you can absolutely have the LEDs in the `g_led_config` array, but not attached to keys. 

Just set the flags as "underglow" for them, and you should be fine. 

Then you can use the `rgb_matrix_indicator_*` function